receiver/otlpreceiver/erroradapter.go

```diff
@@ -0,0 +1,48 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otlpreceiver // import "go.opentelemetry.io/collector/receiver/otlpreceiver"
+
+import (
+	"net/http"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func gRPCToHTTP(s *status.Status) int {
+	switch s.Code() {
+	case codes.Aborted:
+		return http.StatusInternalServerError
+	case codes.AlreadyExists:
+		return http.StatusConflict
+	case codes.Canceled, codes.DataLoss:
```

Comment:
I would put these two on separate lines to keep consistent here (or move all the same codes on single lines)